### PR TITLE
Remove confusing navigation bar from stdlib documentation

### PR DESCRIPTION
**Pull request by johnwhitington**
Here is a page from the current manual, accessed from index.html --> "The Standard Library" --> "Module bool: boolean values":

https://ocaml.org/releases/4.11/htmlman/libref/Bool.html

It contains a navigation bar, added by ocamldoc when ocaml is built (not when the main manual is built), with "Previous", "Up", and "Next" buttons.

Clicking the "Up" button does not take the user back to the standard library list of modules (from where they have just come), but to an otherwise-unlinked ocamldoc index of modules:

https://ocaml.org/releases/4.11/htmlman/libref/index.html

This is unexpected, and confusing -- the user was browsing the reference manual and is now lost.

This PR simply removes the navbar, and so removes this confusion. In practice, the "Up" button is just the "Back" button in the browser, most of the time. The "Previous" and "Next" buttons are of limited use, since browsing the stdlib module by module is an unusual activity, and can be achieved with the browser back button anyway.

If we want to re-instate "Up", and have it point somewhere sensible (i.e back to the list of stdlib modules in the reference manual), this could be done by the HTML post processor https://github.com/ocaml/ocaml/pull/9755

But I think we should get rid of it.

(This PR also includes two tiny HTML manual related commits which were too small to be separate PRs).

(The change to manual/manual/Makefile removes content which is in fact redundant)

### CODE REVIEW

File: manual/manual/Makefile [hunk 1]
```diff
@@ -4,16 +4,8 @@ export LD_LIBRARY_PATH   ?= "$(SRC)/otherlibs/unix/:$(SRC)/otherlibs/str/"
 export DYLD_LIBRARY_PATH ?= "$(SRC)/otherlibs/unix/:$(SRC)/otherlibs/str/"
 SET_LD_PATH = CAML_LD_LIBRARY_PATH=$(LD_LIBRARY_PATH)
 
-OCAMLDOC = $(if $(wildcard $(SRC)/ocamldoc/ocamldoc.opt),\
-  $(SRC)/ocamldoc/ocamldoc.opt,\
-  $(SET_LD_PATH) $(SRC)/runtime/ocamlrun $(SRC)/ocamldoc/ocamldoc)\
-  -hide Stdlib -lib Stdlib -nostdlib \
-  -pp "$(AWK) -v ocamldoc=true -f $(SRC)/stdlib/expand_module_aliases.awk"
```
Comment (dra27) [inline manual/manual/Makefile:11]: This is a clean-up following #9996

File: ocamldoc/odoc_html.ml [hunk 2]
```diff
@@ -2396,7 +2397,7 @@ class html =
         bs b "<html>\n";
         self#print_header b (self#inner_title title);
         bs b "<body>\n";
-        self#print_navbar b None None "";
+        if !show_navbar then self#print_navbar b None None "";
```
Comment (dra27) [inline ocamldoc/odoc_html.ml]: Perhaps put the `if` in `print_navbar` itself (and, if the idea of a function called `print_` which may not print offends, rename the method to `process_navbar`/`do_navbar` or something).

File: ocamldoc/odoc_messages.ml [hunk 3]
```diff
@@ -388,7 +388,7 @@ let authors = "Author(s)"
 let version = "Version"
 let since = "Since"
 let before = "Before"
-let deprecated = "Deprecated."
+let deprecated = "Deprecated. "
```
Comment (dra27) [inline ocamldoc/odoc_messages.ml]: This adds unwanted space in a few places, e.g.:
https://github.com/ocaml/ocaml/blob/59320b5c08f4879344a901e4c4c2a4888b551e39/ocamldoc/odoc_texi.ml#L532
and
https://github.com/ocaml/ocaml/blob/trunk/ocamldoc/odoc_man.ml#L209-L210
I think the fix needs to be here instead:
https://github.com/ocaml/ocaml/blob/59320b5c08f4879344a901e4c4c2a4888b551e39/ocamldoc/odoc_html.ml#L746-L747
Incidentally, this usage looks wrong (generating `Deprecated.!`, but I didn't dig further as to where:
https://github.com/ocaml/ocaml/blob/59320b5c08f4879344a901e4c4c2a4888b551e39/ocamldoc/odoc_misc.ml#L220

**Review by dra27 — COMMENTED** [review]
The `-nonavbar` option to ocamldoc warrants a `Changes` entry, along with a note that it's being used (if someone happens to miss it, they may look in Changes to see why)

**Comment by johnwhitington** [comment]
Added Changes twice, with differing wording: once it Tools, to mention ocamldoc, once in Documentation, to mention effect.

File: ocamldoc/odoc_messages.ml (hunk 3 above)
Comment (johnwhitington) [reply ocamldoc/odoc_messages.ml]: I have changed the message to "Deprecated" and updated other places to add the full stop and/or space. I have checked the HTML, text, and PDF output and they are now correct.

**Comment by johnwhitington** [comment]
All changes requested made.

File: ocamldoc/odoc_misc.ml [hunk 4]
```diff
@@ -217,7 +217,7 @@ let string_of_info i =
   let module M = Odoc_types in
   (match i.M.i_deprecated with
     None -> ""
-  | Some d -> Odoc_messages.deprecated^"! "^(string_of_text d)^"\n")^
+  | Some d -> Odoc_messages.deprecated^".! "^(string_of_text d)^"\n")^
```
Comment (dra27) [inline ocamldoc/odoc_misc.ml]: Does this end up in plain text output (i.e. is a typo) or is it something deeper?

Comment (johnwhitington) [reply ocamldoc/odoc_misc.ml]: "Deprecated.!" is not found in the manual.txt output.

I don't understand what's going on here but I have not changed the net result: if this code was wrong (or right) before, it's wrong (or right) now.

Comment (dra27) [reply ocamldoc/odoc_misc.ml]: If it's not in the plain text manual (and so not really testable), then I agree with leaving it (weird as it looks!)

**Review by dra27 — APPROVED** [review]
LGTM. @Octachron ?

File: ocamldoc/odoc_misc.ml (hunk 4 above)
Comment (Octachron) [reply ocamldoc/odoc_misc.ml]: This is wrong, but this only appears on a difficult to trigger path: this is only visible on the texi output for inline record fields:

```ocaml
type t =
| A of { x:int (** A.x @deprecated "A.x" *); y :int } (** A @deprecated "A" *)
| B
```
is rendered as
```
  | A of @{
   x : int;(* Deprecated.! "A.x"
A.x
 *)
   y : int;
@}     
     (*
     Deprecated. "A"

A

 *)
  | B
```

**Comment by Octachron** [comment]
This seems fine to me (even if I use the next link during reviews from time to time, this is not exactly a feature for most users). Moreover, this navigation bar is also removed by #9755 .

File: ocamldoc/odoc_misc.ml (hunk 4 above)
Comment (johnwhitington) [reply ocamldoc/odoc_misc.ml]: So I can just remove the exclamation mark?

Comment (Octachron) [reply ocamldoc/odoc_misc.ml]: Indeed.

**Comment by johnwhitington** [comment]
Ok, all done.

**Pull request merged**